### PR TITLE
SONARRUBY-37 update slang version to 1.17.0.6351

### PR DESCRIPTION
**Pull request by erwan-serandour**
[SONARRUBY-37](https://sonarsource.atlassian.net/browse/SONARRUBY-37)

Part of 

[SONARRUBY-37]: https://sonarsource.atlassian.net/browse/SONARRUBY-37?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -34,10 +34,12 @@ dependencyResolutionManagement {
...
             library("slang-testing", "org.sonarsource.slang", "slang-testing").version(slangDependenciesVersion)
-            library("mockito-core", "org.mockito", "mockito-core").version("5.13.0")
+            library("mockito-core", "org.mockito", "mockito-core").version("5.14.2")
             library("assertj-core", "org.assertj", "assertj-core").version("3.26.3")
-            library("junit-jupiter-api", "org.junit.jupiter", "junit-jupiter-api").version("5.11.0")
-            library("junit-jupiter-engine", "org.junit.jupiter", "junit-jupiter-engine").version("5.11.0")
+            library("junit-jupiter-api", "org.junit.jupiter", "junit-jupiter-api").version("5.11.3")
+            library("junit-jupiter-engine", "org.junit.jupiter", "junit-jupiter-engine").version("5.11.3")
+            library("junit", "junit", "junit").version("4.13.2")
+            //testImplementation group: 'junit', name: 'junit', version: '4.13.2'
```
Comment (dorian-burihabwa-sonarsource) [inline settings.gradle]: Can we make these library upgrades in a separate PR? Just to make sure we don't run into any issue before release?

**Review by dorian-burihabwa-sonarsource — APPROVED** [review]

File: settings.gradle (hunk 1 above)
Comment (erwan-serandour) [reply settings.gradle]: it is done, I forgot to remove them with the others
